com.unity.multiplayer.mlapi/Prototyping/NetworkAnimator.cs

```diff
@@ -12,235 +12,447 @@ namespace MLAPI.Prototyping
+        {
+            if (IsAuthorityOverAnimator)
             {
-                m_TrackedParamFlags &= (uint)~(1 << paramIndex);
+                m_ServerRequestsAnimationResync = true;
             }
+            
+            var clientRpcParams = new ClientRpcParams
+            {
+                Send = new ClientRpcSendParams
+                {
+                    TargetClientIds = NetworkManager.ConnectedClientsList
+                        .Where(c => c.ClientId != NetworkManager.ServerClientId)
+                        .Select(c => c.ClientId)
+                        .ToArray()
+                }
+            };
+                
+            RequestResyncClientRpc(clientRpcParams);
         }
 
-        public bool GetParamTracking(int paramIndex)
+
+        [ClientRpc]
+        private void RequestResyncClientRpc(ClientRpcParams clientRpcParams = default)
         {
-            if (paramIndex >= 32)
+            if (!IsAuthorityOverAnimator)
             {
-                return false;
+                return;
             }
 
-            return (m_TrackedParamFlags & (uint)(1 << paramIndex)) != 0;
+            m_ServerRequestsAnimationResync = true;
         }
 
-        public void ResetTrackedParams()
-        {
-            m_TrackedParamFlags = 0;
-        }
+
+        private bool IsAuthorityOverAnimator => (IsClient && AnimatorAuthority == Authority.Owner && IsOwner) || (IsServer && AnimatorAuthority == Authority.Server);
 
         private void FixedUpdate()
         {
-            if (!IsOwner)
+            if (!NetworkObject.IsSpawned)
             {
                 return;
             }
 
-            if (CheckSendRate())
+            if (IsAuthorityOverAnimator)
             {
-                SendTrackedParams();
-            }
-
-            if (CheckStateChange(out int animStateHash, out float animStateTime))
-            {
-                SendAllParamsAndState(animStateHash, animStateTime);
+                bool shouldSendBasedOnTime = CheckSendRate();
+                bool shouldSendBasedOnChanges = StoreState();
+                if (m_ServerRequestsAnimationResync || shouldSendBasedOnTime || shouldSendBasedOnChanges)
+                {
+                    SendAllParamsAndState();
+                    m_AnimatorSnapshot.TriggerParameters.Clear();
+                    m_ServerRequestsAnimationResync = false;
+                }
             }
         }
-
-        private float m_NextSendTime = 0.0f;
-
+        
         private bool CheckSendRate()
         {
             var networkTime = NetworkManager.NetworkTime;
-            if (SendRate != 0 && m_NextSendTime < networkTime)
+            if (m_SendRate != 0 && m_NextSendTime < networkTime)
             {
-                m_NextSendTime = networkTime + SendRate;
+                m_NextSendTime = networkTime + m_SendRate;
                 return true;
             }
 
             return false;
         }
```

Comment:
do we still want to control animator sendrate indepently?

Reply:
I essentially opted to keep the original implementation for that bit - had no opinion on it.